src/component/UserFolderList.js

```diff
@@ -0,0 +1,52 @@
+import "../css/userFolderList.css";
+import FolderPlusImg from "../asset/add.svg";
+import { getFolder } from "../api";
+
+import { useState } from "react";
+
+export function UserFolderList({ names, onFolderClick }) {
+  const [selectedFolderId, setSelectedFolderId] = useState("");
+
+  const handleFolderButtonClick = (e) => {
+    const folderId = e.currentTarget.value;
+    setSelectedFolderId(folderId);
+
+    // 여기서 선택된 폴더 아이디로 API를 호출
+    getFolder(folderId)
+      .then((result) => {
+        const { data } = result;
+        onFolderClick(data, true);
+        console.log(data, "받아온값");
+      })
+      .catch((error) => {
+        console.error("Error fetching folder data:", error);
```

Comment:
이 경우에는 사용자는 어찌된 일인지 알 수가 없겠네요. 아직 다이얼로그를 구현하지않았다면, alert도 좋을거같아요

(실무적으로는 log나 error를 UI수준에서 직접 찍는 경우가 드물답니다. )


```suggestion
       alert(`폴더 데이터를 가져올 수 없습니다 ${error.message}`);
```